### ISSUE TITLE
Sync Batch Norm hangs if i don't call a module in one of the Processes

### ISSUE THREAD

**Issue by soulslicer** [pytorch/pytorch#44957]
I have a special model where, during the first iterate of a video, one of the components doesnt get used for example:

```
if self.iter != 0:
    stuff = self.modelA(input)
    stuff2 = self.modelB(stuff)
    return stuff2
else:
    stuff = self.modelA(input)
    return stuff
```

One of the GPU's / process might hit self.iter = 0, while another GPU may still be running stuff with self.iter = N. When this happens, the model hangs on any batch norm.

How can I get around this?

**Comment by SsnL** [comment]
Syncbn, as its name suggests, synchronizes (and thus waits) across all ranks in the process group. All ranks must call this op the same number of times.

**Comment by soulslicer** [comment]
What happens if I "fake" call it by simply calling it with a zero tensor, but I do not return it to be part of the output or loss function, so its not part of the graph. I assume this wont work either?

**Comment by soulslicer** [comment]
Nevermind, since it so happens that my model is dealing with probability distributions, when i have iteration 0, i shall just pass in a uniform distribution so the module gets called always
